src/screens/time/PunchRequestSuccess.tsx

```diff
@@ -0,0 +1,522 @@
+/*
+ * This file is part of OrangeHRM
+ *
+ * Copyright (C) 2020 onwards OrangeHRM (https://www.orangehrm.com/)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import React from 'react';
+import {View, StyleSheet} from 'react-native';
+import {
+  NavigationProp,
+  ParamListBase,
+  StackActions,
+  DrawerActions,
+} from '@react-navigation/native';
+import MainLayout from 'layouts/MainLayout';
+import withTheme, {WithTheme} from 'lib/hoc/withTheme';
+import {connect, ConnectedProps} from 'react-redux';
+import {RootState} from 'store';
+import {
+  fetchPunchStatus,
+  changePunchCurrentDateTime,
+  setPunchNote,
+  savePunchInRequest,
+  savePunchOutRequest,
+} from 'store/time/attendance/actions';
+import {PunchRequestSuccessParam} from 'screens/time/navigators/index';
+import Text from 'components/DefaultText';
+import Divider from 'components/DefaultDivider';
+import Icon from 'components/DefaultIcon';
+import {PunchRequestSuccessRouteParams} from 'screens/time/navigators';
+import IconButton from 'components/DefaultIconButton';
+import {
+  calculateDurationUsingSavedFormat,
+  formatLastRecordDetails,
+} from 'lib/helpers/attendance';
+import {selectInitialRoute} from 'store/globals/selectors';
+
+class PunchRequestSuccess extends React.Component<PunchRequestSuccessProps> {
+  constructor(props: PunchRequestSuccessProps) {
+    super(props);
+  }
+
+  onClickHomeButton = () => {
+    const {initialRoute, navigation} = this.props;
+    navigation.dispatch(StackActions.pop());
+    navigation.dispatch(DrawerActions.jumpTo(initialRoute));
+  };
+  render() {
+    let PUNCH_IN = 'PUNCH_IN';
+    let PUNCH_OUT = 'PUNCH_OUT';
+    let ACTION: typeof PUNCH_IN | typeof PUNCH_OUT;
+    const {theme, route} = this.props;
+
+    const {
+      punchInDateTime,
+      punchInTimeZoneOffset,
+      punchInNote,
+      punchOutDateTime,
+      punchOutTimeZoneOffset,
+      punchOutNote,
+      datetime,
```

Comment:
because response variables are different in two cases
eg: Punch In success     note
      Punch Out success   punchInNote and punchOutNote.
  same for other variables.